test/OpenFeature.Contrib.Providers.Flagsmith.Test/FlagsmithProviderTest.cs

```diff
@@ -0,0 +1,432 @@
+            var glossDiv = glossary.AsStructure.GetValue("GlossDiv");
+            Assert.True(glossDiv.IsStructure);
+            var glossList = glossDiv.AsStructure.GetValue("GlossList");
+            Assert.True(glossList.IsStructure);
+            var glossEntry = glossList.AsStructure.GetValue("GlossEntry");
+            Assert.True(glossEntry.IsStructure);
+            Assert.Equal("SGML", glossEntry.AsStructure.GetValue("SortAs").AsString);
+            var glossDef = glossEntry.AsStructure.GetValue("GlossDef");
+            Assert.True(glossDef.IsStructure);
+            var glossSeeAlso = glossDef.AsStructure.GetValue("GlossSeeAlso");
+            Assert.True(glossSeeAlso.IsList);
+            Assert.Equal(2, glossSeeAlso.AsList.Count);
+            Assert.Equal("GML", glossSeeAlso.AsList.First().AsString);
+            Assert.Equal("XML", glossSeeAlso.AsList.Last().AsString);
+
+            Assert.Equal("example-feature", result.FlagKey);
+            Assert.Null(result.Reason);
+            Assert.Equal(ErrorType.None, result.ErrorType);
+        }
+
+        [Fact]
+        public async Task GetStructureValue_ForDisabledFeatureWithValidFormat_ReturnDefaultValue()
+        {
+            // Arrange
+            var flagsmithClient = Substitute.For<IFlagsmithClient>();
+            var flags = Substitute.For<IFlags>();
+            flags.GetFeatureValue("example-feature").Returns("4112");
+            flags.IsFeatureEnabled("example-feature").Returns(false);
+            flagsmithClient.GetEnvironmentFlags().Returns(flags);
+
+            var defaultObject = new Value("default");
+            var flagsmithProvider = new FlagsmithProvider(flagsmithClient);
+
+            // Act
+            var result = await flagsmithProvider.ResolveStructureValue("example-feature", defaultObject);
+
+            // Assert
+            Assert.Equal(defaultObject, result.Value);
+            Assert.Equal("example-feature", result.FlagKey);
+            Assert.Equal(Reason.Disabled, result.Reason);
+            Assert.Equal(ErrorType.None, result.ErrorType);
+        }
+
+        [Fact]
+        public async Task GetStructureValue_ForEnabledFeatureWithWrongFormatValue_ReturnDefaultValue()
+        {
+            // Arrange
+            var flagsmithClient = Substitute.For<IFlagsmithClient>();
+            var flags = Substitute.For<IFlags>();
+            flags.GetFeatureValue("example-feature").Returns("hreni");
+            flags.IsFeatureEnabled("example-feature").Returns(true);
+            flagsmithClient.GetEnvironmentFlags().Returns(flags);
+
+            var defaultObject = new Value("default");
+            var flagsmithProvider = new FlagsmithProvider(flagsmithClient);
+
+            // Act and Assert
+            await Assert.ThrowsAsync<TypeMismatchException>(() => flagsmithProvider.ResolveStructureValue("example-feature", defaultObject));
+        }
+    }
+
+    public class ExampleConfig
```

Comment:
I can't see where this class is actually being used?

Reply:
I just forgot to delete. Thanks)